app/src/main/java/com/konarsubhojit/synckro/providers/onedrive/OneDriveAuthManager.kt

```
@@ -3,49 +3,317 @@ package com.konarsubhojit.synckro.providers.onedrive
+                            msalApp = application
+                            Timber.d("OneDriveAuthManager: MSAL app created successfully")
+                            cont.resume(application)
+                        }
+
+                        override fun onError(exception: MsalException?) {
+                            Timber.e(exception, "OneDriveAuthManager: Failed to create MSAL app")
+                            cont.resume(null)
+                        }
+                    },
+                )
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)